galaxy_ng/app/api/utils.py

```diff
@@ -53,3 +58,31 @@ def parse_collection_filename(filename):
         raise ValueError(f"Expected version to be max length of {MAX_LENGTH_VERSION}")
 
     return CollectionFilename(namespace, name, version)
+
+
+class SocketHTTPConnection(HTTPConnection):
+    def __init__(self, socket_file):
+        self.socket_file = socket_file
+        super().__init__(LOCALHOST)
+
+    def connect(self):
+        self.sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
+        self.sock.connect(self.socket_file)
```

Comment:
Does HTTPConnection() take care of closing self.sock?  (probably, but haven't verified).

Not sure if it would be an issue even without the close, but maybe worth verifying